### DISCUSSION TITLE
Highlight Edges in IFC?

### DISCUSSION THREAD

**Discussion by neostim** [IfcOpenShell/IfcOpenShell#3629] [Q&A]
Is there a method within IfcOpenShell that would allow me to highlight all the edges in an IFC, say in the color red? Or only show it in wireframe?

I'm using IFC models to overlay them in Augmented Reality over the real object however I'm trying to figure out how to highlight all of the edges (holes, edges of a steel beam etc..) so they are more visible, or simply show it as a wireframe.

**Comment by aothms** [comment]
When you say "within IfcOpenShell" what do you precisely mean. We are a software module with a wide ecosystem of tools. 

The good news is that since our geometry interpretation is based on a BRep data model (open cascade) we have accurate information on which of the edges are significant wrt to the original IFC input. We don't rely on triangulation for example for the evaluation of boolean operations.

**Reply by neostim** [reply]
Thanks for the reply! I suppose I don't know myself, my end goal is to find a way to automate the process of converting IFC models to be wireframe only, where only the edges are highlighted / visible.

I was hoping there might be a starting point using IFCOpenShell-Python, I see there is mention of edges here: https://blenderbim.org/docs-python/ifcopenshell-python/geometry_processing.html but I'm not too sure where to go from there, I don't see any process of outlining only the edges of the model...but that could be my inexperience.

**Reply by aothms** [reply]
Well yes, that's about it. get_edges() gives you pairs (list of length 2) of indices into the vector of coordinates. So that you have a bunch of line segments.

If you don't want to do programming. You can look into BlenderBIM or do `from ifcopenshell.geom.app import application; application().start()` both have ways of showing a wireframe. The 2nd is not really actively developed though.
